proxy/http/HttpSM.cc

```diff
@@ -7988,3 +7987,58 @@ HttpSM::find_proto_string(HTTPVersion version) const
   }
   return nullptr;
 }
+
+// YTS Team, yamsat Plugin
+// Function to copy the partial Post data while tunnelling
+void
+PostDataBuffers::copy_partial_post_data()
+{
+  this->postdata_copy_buffer->write(this->ua_buffer_reader);
+  Debug("http_redirect", "[PostDataBuffers::copy_partial_post_data] wrote %" PRId64 " bytes to buffers %" PRId64 "",
+        this->ua_buffer_reader->read_avail(), this->postdata_copy_buffer_start->read_avail());
+  this->ua_buffer_reader->consume(this->ua_buffer_reader->read_avail());
+}
+
+// YTS Team, yamsat Plugin
+// Allocating the post data buffers
+void
+PostDataBuffers::init(IOBufferReader *ua_reader)
+{
+  // the this->ua_buffer_reader may be realeased in tunnel::deallocate_buffers.
+  // we always set a new reader instead of the old value;
+  // ink_assert(this->ua_buffer_reader == nullptr);
+
+  Debug("http_redirect", "[PostDataBuffers::init]");
+
+  this->ua_buffer_reader = ua_reader;
+
+  if (this->postdata_copy_buffer == nullptr) {
+    ink_assert(this->postdata_copy_buffer_start == nullptr);
+    this->postdata_copy_buffer       = new_MIOBuffer(BUFFER_SIZE_INDEX_4K);
```

Comment:
IMO, `new_empty_MIOBuffer()` is more better, the same as `postdata_producer_buffer` above.